src/Discord.Net.SlashCommands/SlashCommandContext.cs

```diff
@@ -0,0 +1,37 @@
+namespace Discord.SlashCommands
+{
+    /// <inheritdoc cref="ISlashCommandContext"/>
+    public class SlashCommandContext : ISlashCommandContext
+    {
+        /// <inheritdoc/>
+        public virtual IDiscordClient Client { get; }
+
+        /// <inheritdoc/>
+        public virtual IGuild Guild { get; }
+
+        /// <inheritdoc/>
+        public virtual IMessageChannel Channel { get; }
+
+        /// <inheritdoc/>
+        public virtual IUser User { get; }
+
+        /// <inheritdoc/>
+        public virtual IDiscordInteraction Interaction { get; }
+
+        /// <summary>
+        /// Whether the origin channel of the Interaction is a private channel
+        /// </summary>
+        public bool IsPrivate => Channel is IPrivateChannel;
+
+        public InteractionType InteractionType => Interaction.Type;
+
+        public SlashCommandContext (IDiscordClient client, IMessageChannel channel, IUser user, IDiscordInteraction interaction)
```

Comment:
Maybe the channel and user can be extrapolated from interaction instead of relying on the user to specify them

Reply:
Also in case of context commands the channel would be null